File: CS/E4422/CustomRichEdit/CustomEdit.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.ComponentModel;
using DevExpress.XtraEditors;

namespace CustomEditor {
    public class CustomEdit : RichTextEdit {
        //The static constructor which calls the registration method
        static CustomEdit() {
            RepositoryItemCustomEdit.RegisterCustomEdit();
        }

        //Initialize the new instance
        public CustomEdit() {
            //...
        }

        //Return the unique name
        public override string EditorTypeName {
            get { return RepositoryItemCustomEdit.CustomEditName; }
        }

        //Override the Properties property
        //Simply type-cast the object to the custom repository item type
        [DesignerSerializationVisibility(DesignerSerializationVisibility.Content)]
        public new RepositoryItemCustomEdit Properties {
            get { return base.Properties as RepositoryItemCustomEdit; }
        }
    }
}

File: VB/E4422/CustomRichEdit/CustomEdit.vb
Imports System.ComponentModel
Imports DevExpress.XtraEditors

Namespace CustomEditor

    Public Class CustomEdit
        Inherits RichTextEdit

        'The static constructor which calls the registration method
        Shared Sub New()
            RepositoryItemCustomEdit.RegisterCustomEdit()
        End Sub

        'Initialize the new instance
        Public Sub New()
        '...
        End Sub

        'Return the unique name
        Public Overrides ReadOnly Property EditorTypeName As String
            Get
                Return RepositoryItemCustomEdit.CustomEditName
            End Get
        End Property

        'Override the Properties property
        'Simply type-cast the object to the custom repository item type
        <DesignerSerializationVisibility(DesignerSerializationVisibility.Content)>
        Public Overloads ReadOnly Property Properties As RepositoryItemCustomEdit
            Get
                Return TryCast(MyBase.Properties, RepositoryItemCustomEdit)
            End Get
        End Property
    End Class
End Namespace
